### PR TITLE
Partially inline mono_method_signature_checked/internal.

### PR DESCRIPTION
**Pull request by jaykrell**
Or consider LTO/LTCG on desktop.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/profiler/coverage.c [hunk 1]
```diff
@@ -278,7 +278,7 @@ dump_method (gpointer key, gpointer value, gpointer userdata)
 	image = mono_class_get_image (klass);
 	image_name = mono_image_get_name (image);
 
-	method_signature = mono_signature_get_desc (mono_method_signature_internal (method), TRUE);
+	method_signature = mono_signature_get_desc (mono_method_signature_profiler (method), TRUE);
```
Comment (lambdageek) [inline mono/profiler/coverage.c]: Why can't the profiler use the `static inline` definitions?

Comment (jaykrell) [reply mono/profiler/coverage.c]: Initially I had separate out of line profiler functions and profiler couldn’t use unexporteds. Now, agreed, should work, will apply that. Thank you/sorry.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**